src/asm.rs

```diff
@@ -15,6 +15,30 @@ pub fn bkpt() {
     }
 }
 
+/// Blocks the program for at least `n` instruction cycles
+///
+/// This is implemented in assembly so its execution time is the same regardless of the optimization
+/// level.
+///
+/// NOTE that the delay can take much longer if interrupts are serviced during its execution.
+#[inline(never)]
```

Comment:
Without `#[inline(never)]` the compiler seems to misoptimize this function: it doesn't reload the `$0` register on every invocation but instead it starts decrementing it from its previous value (0).

Reply:
This is not misoptimized, it is intended behavior. You used the "r" constraint and promised to the compiler that you will not modify the value in this register. There is a way to fix it, use "r+" modifier to let compiler know that you will write this input register. Or you can copy the value in some other register, add it to the clobber list and modify it instead of input register.